### PR TITLE
Refactored app to use async/await

### PR DESCRIPTION
**Pull request by buigabor**
Refactored the whole app to use async/await

### CODE REVIEW

File: public/js/index.js [hunk 1]
```diff
@@ -128,250 +133,196 @@ function handleViewClick(lesson) {
...
+			.doc(uid)
+			.set({ uid: uid, lessons: [] })
+			.catch((error) => {
+				console.error('Error adding document: ', error);
+			});
+	}
 }
 
-function addLesson(e) {
-	e.preventDefault();
```
Comment (jrobind) [inline public/js/index.js:275]: Is there a reason why `e.preventDefault();` has been removed from the function? I can see you've moved it into an anonymous function on line 29.

File: public/js/index.js [hunk 2]
```diff
@@ -128,250 +133,196 @@ function handleViewClick(lesson) {
 	overlay.classList.add('dark');
 }
 
-function lessonHelper({varName, eventListener, classList, attribute, textContent, innerHTML, id}) {
```
Comment (jrobind) [inline public/js/index.js:131]: I don't think we want to remove this helper function. Not sure if this was intentional or not.

**Review by jrobind — CHANGES_REQUESTED** [review]
@buigabor firstly, awesome job. This is pretty advanced JS so 👏 

The async await logic looks great, I've just left a few small comments that need addressing before merging 😄 

Thanks!

File: public/js/index.js (hunk 1 above)
Comment (buigabor) [reply public/js/index.js:275]: Yeah, so e.preventDefault() did not work when I refactored this function into async/await. So I looked it up and followed this guideline:

https://github.com/github/eslint-plugin-github/blob/main/docs/rules/async-preventdefault.md

Comment (jrobind) [reply public/js/index.js:275]: That makes sense. Good to know too, thanks for sharing!

File: public/js/index.js (hunk 2 above)
Comment (buigabor) [reply public/js/index.js:131]: Oh god, I may have deleted it accidentally. I will re-add it.

Comment (buigabor) [reply public/js/index.js:131]: I re-added it with the following commit: 74fb323c6b210cca11ba958f1ea9206b14279fb1

File: public/js/index.js [hunk 3]
```diff
@@ -117,261 +122,247 @@ function lessonHandler(e) {
...
-		.get()
-		.then((snapshot) => {
-			if (lessonsContainer.childElementCount) {
-				lessonsContainer.innerHTML = '';
-			}
-
-			if (snapshot.data() && snapshot.data().lessons.length) {
-				snapshot.data().lessons.forEach(({ title, content, id }) => {
-					const lessonCard = lessonHelper({
-						varName: document.createElement('div'),
```
Comment (jrobind) [inline public/js/index.js:172]: One last thing 😛  can you add these `lessonHelper` function calls back in please. I think these may have been removed by mistake too. Then we are good to go 👍

**Review by jrobind — CHANGES_REQUESTED** [review]

**Review by jrobind — APPROVED** [review]
Perfect, thanks!

**Pull request merged**